pkg/ds/stack/stack.go

```diff
@@ -0,0 +1,52 @@
+package stack
+
+// Stack is a generic stack.
+type Stack[T any] struct {
+	elems []T
+}
+
+// New returns a new instance of Stack.
+// hint: hint for stack size.
+func New[T any](hint int) *Stack[T] {
+	if hint < 0 {
+		hint = 0
+	}
+
+	return &Stack[T]{elems: make([]T, 0, hint)}
+}
+
+// Empty returns true if the stack is empty.
+func (s *Stack[T]) Empty() bool {
+	return len(s.elems) == 0
+}
+
+// Len returns number of elements in the stack.
+func (s *Stack[T]) Len() int {
+	return len(s.elems)
+}
+
+// Peek views the top element of the stack.
+// nolint: ireturn
+func (s *Stack[T]) Peek() T {
```

Comment:
Should check empty stack

Reply:
Fixed https://github.com/KyberNetwork/tradinglib/pull/3/commits/26b7d6688a515421dd3d532c19f130cc5a9ba03f